src/azul/service/avro_pfb.py

```diff
@@ -602,3 +639,122 @@ def _entity_schema_recursive(field_types: FieldTypes,
             pass
         else:
             assert False, field_type
+
+
+def _update_replica_schema(*,
+                           schema: MutableJSON,
+                           path: Sequence[str],
+                           key: str,
+                           value: AnyMutableJSON):
+    """
+    Update in place a (part of an) existing PFB schema to ensure that it
+    accommodates a given (part of a) JSON document. The schema will only ever
+    expand, so after updating it will describe a superset of the documents that
+    it described pre-update. Starting from an empty schema, repeatedly calling
+    this function this allows us to discover a general schema for a series of
+    documents of unknown shape.
+    :param schema: a part of a PFB schema. It may be empty.
```

Comment:
```suggestion

        :param schema: a part of a PFB schema. It may be empty.
```

And between the individual parameters, too, as they are all separate paragraphs. In all docstrings that you add.

Reply:
The addition of the docstring should occur in the commit that adds the method.